CDAP-18010: Renaming ArtifactInspector to DefaultArtifactInspector

wyzhang opened the pull request:
Why:
Plan to introduce an interfact ArtifactInspector in a follow up change
so it can have a default implementation and remote execution implementation
which allows this to run in isolation (e.g. in task worker k8s pod),
thus better security and potentially more scalable.

tivv commented on cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/runtime/artifact/ArtifactInspectorTest.java:
```diff
@@ -59,21 +59,22 @@
 /**
+ *
  */
 public class ArtifactInspectorTest {
```
nit: rename the test

wyzhang replied:
Done. Fixed

chtyim commented:
You actually can do the refactoring in one shot using the "Extra Interface" in IntelliJ.

<img width="728" alt="Screen Shot 2021-05-26 at 9 37 08 AM" src="https://user-images.githubusercontent.com/885032/119698509-27b22680-be06-11eb-9c30-3c1ba1b24ed8.png">
<img width="721" alt="Screen Shot 2021-05-26 at 9 37 58 AM" src="https://user-images.githubusercontent.com/885032/119698517-2aad1700-be06-11eb-9aa7-68c0b09313eb.png">

wyzhang commented:
Thanks @chtyim.  Will use that in future.

wyzhang committed:
CDAP-18010: Renaming ArtifactInspector to DefaultArtifactInspector

Why:
Plan to introduce an interfact ArtifactInspector in a follow up change
so it can have a default implementation and remote execution implementation
which allows this to run in isolation (e.g. in task worker k8s pod),
thus better security and potentially more scalable.
